File: CompilationTest/MapKeyTest/results/MapKeyTest.cpp
#include <OneLang-Core-v0.1/one.hpp>
#include <map>
#include <string>

class TestClass {
  public:
    void testMethod() {
        auto map = one::make_shared_map<std::string, any>({
        });
        auto keys = OneMapHelper::keys(map);
    }

  private:
};

int main()
{
    try {
        TestClass c;
        c.testMethod();
    } catch(std::exception& err) {
        std::cout << "Exception: " << err.what() << '\n';
    }
    return 0;
}

File: CompilationTest/MapKeyTest/results/MapKeyTest.go
package main

import "fmt"

type TestClass struct {
}

func NewTestClass() *TestClass {
    this := new(TestClass)
    return this
}

func (this *TestClass) TestMethod() {
    map := map[string]interface{}{
    }
    keys := make([]string, 0, len(map))
    for  key, _ := range map {
      keys = append(keys, key)
    }
}

func main() {
    defer func() {
      if r := recover(); r != nil {
          fmt.Print("Exception: ", r)
      }
    }()

    c := NewTestClass()
    c.TestMethod();
}